src/main/java/net/snowflake/ingest/streaming/internal/StreamingIngestStage.java

```diff
@@ -350,6 +355,14 @@ private JsonNode parseClientConfigureResponse(Map<String, Object> response) {
 
     // JDBC expects this field which maps to presignedFileUrlName.  We will set this later
     dataNode.putArray("src_locations").add("placeholder");
+    if (expectedDeploymentId != null) {
+      Long actualDeploymentId =
+          responseNode.has("deployment_id") ? responseNode.get("deployment_id").longValue() : null;
```

Comment:
Now I'm thinking that we have the assumption that the deployment id for a deployment will never change, is that a correct assumption?

Reply:
Asked on #metadata-issues-discuss to confirm.

Reply:
Confirmed.